### PR TITLE
made the lnsocket library a module and then imported said module

### PR DESCRIPTION
**Pull request by kevkevinpal**
Sorry in advanced my prettier auto formatter went rouge and formatted everything but I can add comments to the things I changed

basically added this npm module
https://www.npmjs.com/package/lnsocketbykevin

The code is in this repo
https://github.com/kevkevinpal/lnsocket

I can show how to publish the node module since I wouldn't be the best maintainer since idk what it does

I also removed the code from index.html so we don't inject it in the code and instead include it as an package.json dependency

and then I import it in the component

Sorry this was kinda rushed I can take another look over the weekend

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,76 +1,77 @@
...
+        "@ng-bootstrap/ng-bootstrap": "4.2.1",
+        "@ngx-translate/core": "11.0.1",
+        "@ngx-translate/http-loader": "4.0.0",
+        "angular-gauge-chart": "^0.7.2",
+        "bip32": "^3.0.1",
+        "bitcoinjs-lib": "^6.0.1",
+        "chart.js": "2.7.3",
+        "core-js": "2.5.7",
+        "ecpair": "^2.0.1",
+        "lnsocketbykevin": "^1.0.2",
```
Comment (kevkevinpal) [inline package.json:34]: Here is the new package I imported

File: src/index.html [hunk 2]
```diff
@@ -6,9 +6,12 @@
...
 
         <meta name="viewport" content="width=device-width, initial-scale=1" />
-        <link rel="icon" type="image/x-icon" href="assets/images/logoicon.png" />
+        <link
+            rel="icon"
+            type="image/x-icon"
+            href="assets/images/logoicon.png"
+        />
 
-        <script src="assets/lnsocket.js"></script>
```
Comment (kevkevinpal) [inline src/index.html:11]: deleted here

File: src/app/layout/dashboard/dashboard.component.ts [hunk 3]
```diff
@@ -1,63 +1,66 @@
...
-import { routerTransition } from '../../router.animations';
-
-import { Observable, BehaviorSubject, of } from 'rxjs';
-import { filter, map, tap, flatMap } from 'rxjs/operators';
+import { Component, OnInit, AfterViewInit } from "@angular/core";
+import { routerTransition } from "../../router.animations";
 
+import { Observable, BehaviorSubject, of } from "rxjs";
+import { filter, map, tap, flatMap } from "rxjs/operators";
+import { lnsocket_init } from "lnsocketbykevin";
```
Comment (kevkevinpal) [inline src/app/layout/dashboard/dashboard.component.ts]: Imported new package here

**Comment by niftynei** [comment]
hits an error for me when i run it locally: `fn is not defined` hmm.

```
core.js:7187 ERROR Error: Uncaught (in promise): ReferenceError: fn is not defined
ReferenceError: fn is not defined
    at lnsocket.js:9
    at lnsocket.js:9
    at lnsocket_init (lnsocket.js:24)
    at DashboardComponent.<anonymous> (dashboard.component.ts:25)
    at Generator.next (<anonymous>)
    at tslib.es6.js:74
    at new ZoneAwarePromise (zone-evergreen.js:876)
    at Module.__awaiter (tslib.es6.js:70)
    at DashboardComponent.connectNode (dashboard.component.ts:23)
    at DashboardComponent.<anonymous> (dashboard.component.ts:261)
    at resolvePromise (zone-evergreen.js:797)
    at zone-evergreen.js:707
    at rejected (tslib.es6.js:72)
    at ZoneDelegate.invoke (zone-evergreen.js:359)
    at Object.onInvoke (core.js:30892)
    at ZoneDelegate.invoke (zone-evergreen.js:358)
    at Zone.run (zone-evergreen.js:124)
    at zone-evergreen.js:855
    at ZoneDelegate.invokeTask (zone-evergreen.js:391)
    at Object.onInvokeTask (core.js:30873)
```

**Comment by kevkevinpal** [comment]
> hits an error for me when i run it locally: `fn is not defined` hmm.
> 
> ```
> core.js:7187 ERROR Error: Uncaught (in promise): ReferenceError: fn is not defined
> ReferenceError: fn is not defined
>     at lnsocket.js:9
>     at lnsocket.js:9
>     at lnsocket_init (lnsocket.js:24)
>     at DashboardComponent.<anonymous> (dashboard.component.ts:25)
>     at Generator.next (<anonymous>)
>     at tslib.es6.js:74
>     at new ZoneAwarePromise (zone-evergreen.js:876)
>     at Module.__awaiter (tslib.es6.js:70)
>     at DashboardComponent.connectNode (dashboard.component.ts:23)
>     at DashboardComponent.<anonymous> (dashboard.component.ts:261)
>     at resolvePromise (zone-evergreen.js:797)
>     at zone-evergreen.js:707
>     at rejected (tslib.es6.js:72)
>     at ZoneDelegate.invoke (zone-evergreen.js:359)
>     at Object.onInvoke (core.js:30892)
>     at ZoneDelegate.invoke (zone-evergreen.js:358)
>     at Zone.run (zone-evergreen.js:124)
>     at zone-evergreen.js:855
>     at ZoneDelegate.invokeTask (zone-evergreen.js:391)
>     at Object.onInvokeTask (core.js:30873)
> ```

So I see on line 9 of the lnsocket.js file that `fn` is assigned to `randomValuesNode` or `randomValuesStandard` I'm not sure what those are referring to. Are they some sort of config file?

**Comment by jb55** [comment]
yes this is a bug and I'm surprised I have not hit this 🤔

I have already made an lnsocket npm module for nodejs, but the emscripten-generated code is different for the browser so it's a separate build, I haven't got a chance to make the npm package just work out of the box depending on the environment (node/browser).

it shouldn't be too difficult.. I'll update the npm module soon.

**Comment by kevkevinpal** [comment]
seems for the `fn` issue we had to instantiate it earlier in the code with `let fn` but then I come across another error looking for `lnsocket.wasm`

**Comment by jb55** [comment]
try version 0.3.1 of the lnsocket npm module. Also keep in mind the interface is different:

```js
const LNSocket = require('lnsocket')

async function makeRequest() {
  const ln = await LNSocket()
  ln.genkey()

  // etc
}
```

**Pull request merged**